lifecycle-operator/controllers/lifecycle/keptnappversion/controller_test.go

```diff
@@ -117,6 +117,71 @@ func TestKeptnAppVersionReconciler_reconcile(t *testing.T) {
 
 }
 
+func TestKeptnAppVersionNonBlockingReconciler_Reconcile(t *testing.T) {
+
+	status := apilifecycle.KeptnAppVersionStatus{
+		PreDeploymentTaskStatus: []apilifecycle.ItemStatus{
+			{
+				Name:           "pre-task",
+				DefinitionName: "task",
+				Status:         apicommon.StateFailed,
+			},
+		},
+	}
+
+	appVersionName := fmt.Sprintf("%s-%s", "myapp", "1.0.0")
+	app := &apilifecycle.KeptnAppVersion{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       appVersionName,
+			Namespace:  "default",
+			Generation: 1,
+		},
+		Spec: apilifecycle.KeptnAppVersionSpec{
+			KeptnAppSpec: apilifecycle.KeptnAppSpec{
+				Version: "1.0.0",
+			},
+			KeptnAppContextSpec: apilifecycle.KeptnAppContextSpec{
+				DeploymentTaskSpec: apilifecycle.DeploymentTaskSpec{
+					PreDeploymentTasks: []string{
+						"task",
+					},
+				},
+			},
+			AppName: "myapp",
+			TraceId: map[string]string{
+				"traceparent": "parent-trace",
+			},
+		},
+		Status: status,
+	}
+	r, _, _ := setupReconciler(app)
+	// set up a non blocking deployment
+	r.Config.SetBlockDeployment(false)
+	r.PhaseHandler = &phasefake.MockHandler{HandlePhaseFunc: func(ctx context.Context, ctxTrace context.Context, tracer telemetry.ITracer, reconcileObject client.Object, phaseMoqParam apicommon.KeptnPhaseType, reconcilePhase func(phaseCtx context.Context) (apicommon.KeptnState, error)) (phase.PhaseResult, error) {
+		reconcilePhase(ctx)
```

Comment:
let's check that the return value here

Reply:
I have added an error check to the reconcilePhase statement. Do we want to do anything with the return state in this line? We are checking for it later and therefore, I have discarded it.

Reply:
looks good now, thanks!